src/mlpack/core/tree/octree/octree_impl.hpp

```diff
@@ -363,6 +363,37 @@ Octree<MetricType, StatisticType, MatType>::Octree(const Octree& other) :
   }
 }
 
+//! Copy Assignment
+template<typename MetricType, typename StatisticType, typename MatType>
+Octree<MetricType, StatisticType, MatType>&
+Octree<MetricType, StatisticType, MatType>::
+operator=(const Octree& other)
+{
+  // Return if it's the same tree.
+  if (this == &other)
+    return *this;
+
+  begin = other.Begin();
+  count = other.Count();
+  bound = other.bound;
+  dataset = ((other.parent == NULL) ? new MatType(*other.dataset) : NULL);
+  parent = NULL;
+  stat = other.stat;
+  parentDistance = other.ParentDistance();
+  furthestDescendantDistance = other.FurthestDescendantDistance();
+  metric = other.metric;
+
+  // If we have any children, we need to create them, and then ensure that their
+  // parent links are set right.
+  for (size_t i = 0; i < other.NumChildren(); ++i)
+  {
+    children.push_back(new Octree(other.Child(i)));
```

Comment:
`children` hasn't been cleared before these calls, and also when `children` is cleared we need to be sure that no memory leaks.